dronekit_sitl/__init__.py

```diff
@@ -174,32 +241,35 @@ def launch(self, args, verbose=False, await_ready=False, restart=False, wd=None,
                 raise ChildProcessError('SITL is already running in this process, please use .stop() to kill it')
             self.stop()
 
-        if not wd:
-            wd = tempfile.mkdtemp()
-        self.wd = wd
+        if wd is not None:
+            self.wd = wd
+        if self.wd is None or not use_saved_data:
+            self.wd = tempfile.mkdtemp()
 
-        # Load the binary for primitive feature detection.
-        elf = open(self.path, 'rb').read()
+        if use_saved_data:
+            # make sure there's an eeprom.bin, at least!
+            eeprom_path = os.path.join(self.wd, "eeprom.bin")
+            if not os.path.exists(eeprom_path):
+                raise Exception('Told to use saved data, but (%s) does not exist' % (eeprom_path,))
 
-        # pysim is required for earlier SITL builds
-        # lacking --home or --model params.
-        need_sim = not b'--home' in elf or not b'--model' in elf
-        self.using_sim = need_sim
+        caps = ArdupilotCapabilities(self.path)
```

Comment:
It's possible "capabilities" is the wrong name here entirely!  They're really "what's the interface to the binary we're trying to run", which is dronekit-sitl's domain.  I don't know whether they should be exposed to the outside world.  dronekit-sitl should be exposing things that are applicable to any SITL system we might be abstracting - whether lockstep-scheduling is available, for example, or whether is is possible for the binary to provide visualisation outputs (for FlightGear or whatever).